### PR TITLE
NN-3519 Move prisoner to hospital page

### PR DESCRIPTION
**Pull request by jsrobertson**
image
image

### CODE REVIEW

File: server/views/pages/prisonerSelect.njk [hunk 1]
```diff
@@ -36,7 +36,7 @@
       <a href="{{ pshUrl }}/prisoner/{{ prisoner.prisonerNumber }}" class="govuk-link">{{ prisoner.displayName }}</a>
     {% endset -%}
     {% set moveHospitalLinkHtml %}
-      <a href="/move-to-hospital/{{ prisoner.prisonerNumber }}" class="govuk-link" data-test="prisoner-move-to-hospital-link"><span class="govuk-visually-hidden">{{ prisoner.displayName }} - </span>Move to a hospital</a>
+      <a href="/move-to-hospital/{{ prisoner.prisonerNumber }}?returnUrl=/select-prisoner?searchTerm={{ searchTerm }}" class="govuk-link" data-test="prisoner-move-to-hospital-link"><span class="govuk-visually-hidden">{{ prisoner.displayName }} - </span>Move to a hospital</a>
```
Comment (sp-watson) [inline server/views/pages/prisonerSelect.njk:39]: I would construct this link in the js: IMO putting the returnUrl in the njk makes things more confusing, given that the returnUrl processing code is in JS.

File: server/@types/express/index.d.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 declare module 'express-session' {
   // Declare that the session will potentially contain these additional fields
   export interface SessionData {
+    returnUrl: query
```
Comment (sp-watson) [inline server/@types/express/index.d.ts:4]: I know we have used returnUrl before, but might it be better to use a name that is more representative, like `journeyStartUrl` or `restartUrl`? 

For me, `returnUrl` is used to hold the place to go when a journey/action has completed, e.g. on logon, which isn't the case here

File: server/utils/utils.ts [hunk 3]
```diff
@@ -14,4 +16,9 @@ const properCaseName = (name: string): string => (isBlank(name) ? '' : name.spli
 const convertToTitleCase = (sentence: string): string =>
   isBlank(sentence) ? '' : sentence.split(' ').map(properCaseName).join(' ')
 
+export const addSelect = (selectItems: SelectOption[], text = 'Select'): SelectOption[] => {
+  selectItems.unshift({ value: '', text })
```
Comment (ake3mio) [inline server/utils/utils.ts:20]: Does it make sense to `[{ value: '', text }, ...selectItems]` instead to prevent mutating the input?

Comment (jsrobertson) [reply server/utils/utils.ts:20]: Yeah definitely.  Good spot.

File: server/@types/express/index.d.ts (hunk 2 above)
Comment (jsrobertson) [reply server/@types/express/index.d.ts:4]: Yeah makes sense. Will update.

**Review by sp-watson — APPROVED** [review]

**Pull request merged**